src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/DelayedTaskScheduler.java

```diff
@@ -47,20 +69,34 @@ public void schedule(long time, Runnable task)
 
     public int process()
     {
+        schedulingDeferred = true;
         if (!scheduledTimes.isEmpty())
         {
             long c = System.currentTimeMillis();
             SortedSet<Long> past = scheduledTimes.headSet(c);
-
-            past = new TreeSet<>(past);
-            scheduledTimes.removeAll(past);
-
-            for (Long time : past)
+            Iterator<Long> iter = past.iterator();
+            while(iter.hasNext())
             {
+                Long time = iter.next();
+                iter.remove();
                 taskLookup.remove(time).run();
             }
             return past.size();
```

Comment:
This `return` will prevent `schedulingDeferred` from being assigned back to `false` and deferred tasks from being added until next call to `process`.